src/instrumentations/index.ts

```diff
@@ -15,60 +15,221 @@
+  {
+    module: '@opentelemetry/instrumentation-http',
+    name: 'HttpInstrumentation',
+    shortName: 'http',
+  },
+  {
+    module: '@opentelemetry/instrumentation-ioredis',
+    name: 'IORedisInstrumentation',
+    shortName: 'ioredis',
+  },
+  {
+    module: '@opentelemetry/instrumentation-knex',
+    name: 'KnexInstrumentation',
+    shortName: 'knex',
+  },
+  {
+    module: '@opentelemetry/instrumentation-koa',
+    name: 'KoaInstrumentation',
+    shortName: 'koa',
+  },
+  {
+    module: '@opentelemetry/instrumentation-memcached',
+    name: 'MemcachedInstrumentation',
+    shortName: 'memcached',
+  },
+  {
+    module: '@opentelemetry/instrumentation-mongodb',
+    name: 'MongoDBInstrumentation',
+    shortName: 'mongodb',
+  },
+  {
+    module: '@opentelemetry/instrumentation-mongoose',
+    name: 'MongooseInstrumentation',
+    shortName: 'mongoose',
+  },
+  {
+    module: '@opentelemetry/instrumentation-mysql',
+    name: 'MySQLInstrumentation',
+    shortName: 'mysql',
+  },
+  {
+    module: '@opentelemetry/instrumentation-mysql2',
+    name: 'MySQL2Instrumentation',
+    shortName: 'mysql2',
+  },
+  {
+    module: '@opentelemetry/instrumentation-nestjs-core',
+    name: 'NestInstrumentation',
+    shortName: 'nestjs-core',
+  },
+  {
+    module: '@opentelemetry/instrumentation-net',
+    name: 'NetInstrumentation',
+    shortName: 'net',
+  },
+  {
+    module: '@opentelemetry/instrumentation-pg',
+    name: 'PgInstrumentation',
+    shortName: 'pg',
+  },
+  {
+    module: '@opentelemetry/instrumentation-pino',
+    name: 'PinoInstrumentation',
+    shortName: 'pino',
+  },
+  {
+    module: '@opentelemetry/instrumentation-redis',
+    name: 'RedisInstrumentation',
+    shortName: 'redis',
+  },
+  {
+    module: '@opentelemetry/instrumentation-redis-4',
+    name: 'RedisInstrumentation',
+    shortName: 'redis-4',
+  },
+  {
+    module: '@opentelemetry/instrumentation-restify',
+    name: 'RestifyInstrumentation',
+    shortName: 'restify',
+  },
+  {
+    module: '@opentelemetry/instrumentation-router',
+    name: 'RouterInstrumentation',
+    shortName: 'router',
+  },
+  {
+    module: '@opentelemetry/instrumentation-tedious',
+    name: 'TediousInstrumentation',
+    shortName: 'tedious',
+  },
+  {
+    module: '@opentelemetry/instrumentation-winston',
+    name: 'WinstonInstrumentation',
+    shortName: 'winston',
+  },
+  {
+    module: './external/elasticsearch',
+    name: 'ElasticsearchInstrumentation',
+    shortName: 'elasticsearch',
+  },
+  {
+    module: './external/kafkajs',
+    name: 'KafkaJsInstrumentation',
+    shortName: 'kafkajs',
+  },
+  {
+    module: './external/sequelize',
+    name: 'SequelizeInstrumentation',
+    shortName: 'sequelize',
+  },
+  {
+    module: './external/typeorm',
+    name: 'TypeormInstrumentation',
+    shortName: 'typeorm',
+  },
 ];
 
+function envKey(info: InstrumentationInfo) {
+  return `OTEL_INSTRUMENTATION_${info.shortName.toUpperCase()}_ENABLED` as EnvVarKey;
```

Comment:
Hmm this would lead to instrumentations with - in shortname to be like `OTEL_INSTRUMENTATION_GENERIC-POOL_ENABLED`, right? Maybe for consistency it'd be better to full snakecase (`OTEL_INSTRUMENTATION_GENERIC_POOL_ENABLED`)

looks like it'd have a small ux advantage: doubleclicking on word with only _ would select all of it, while right now it'd break at - (can test with the 2 above)

Reply:
Fixed